### PR TITLE
fix(snippet): update custom field snippet #73

### LINKED ISSUE #73: fix(snippet): update custom field snippet
## User Story

As a Give developer, I want an up-to-date example of how to programmatically add a custom field so that it ends up in post meta like any other custom field.

## Acceptance Criteria

- [x] The `add_post_meta` call has been changed to `give_update_payment_meta` in the snippet.
- [x] The snippet has been tested and confirmed that its post meta is stored correctly in the database.
- [x] The custom field is visible in Donation Details screen.
- [x] The custom field appears on Donation Confirmation page.
- [x] The custom field is available for export in `Tools > Exporter`
- [x] The custom field is available for Email Tags
- [ ] Support team has been notified that the snippet is ready.

## Related

- Snippet URL - https://github.com/WordImpress/Give-Snippet-Library/blob/master/form-customizations/custom-fields-plugin.php
- Outdated doc - https://givewp.com/documentation/developers/how-to-create-custom-form-fields/
- https://github.com/WordImpress/Give-Docs/issues/166

## Tasks

1. Review the [outdated documentation(https://givewp.com/documentation/developers/how-to-create-custom-form-fields/).
2. Update the [snippet](https://github.com/WordImpress/Give-Snippet-Library/blob/master/form-customizations/custom-fields-plugin.php) to work with Give 2.0+
3. Notify support in https://github.com/WordImpress/Give-Docs/issues/166 so the doc can be updated.

### PR DESCRIPTION
**Pull request by raftaar1191**
## Description
PR to fix #73 

## How Has This Been Tested?
Tested by adding custom meta column with some value and without some value

## Task
- [x] Updated `add_post_meta` function with `give_update_payment_meta` in the snippet.
- [x] Tested and confirmed that its post meta is stored correctly in the database.
- [x] Tested that custom field is visible in Donation Details screen.
- [x] Tested custom field appears on Donation Confirmation page.
- [x] Tested custom field is available for export in `Tools > Exporter`
- [x] Tested custom field is available for Email Tags
- [ ] Support team has been notified that the snippet is ready.

## Types of changes
New feature (non-breaking change which adds functionality)

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

**Comment by raftaar1191** [comment]
@kevinwhoffman @mehul0810 @Benunc 

I have updated the snipped with newly updated code. Kindly review this PR and merge. And also update the [Docs](https://github.com/WordImpress/Give-Docs/issues/166)

File: form-customizations/custom-fields-plugin.php [hunk 1]
```diff
@@ -116,7 +117,7 @@ function myprefix123_give_donations_donation_details( $payment_id ) {
 
 	if ( $engraving_message ) : ?>
 
-		<div id="give-engraving-details" class="postbox">
+		<div id="<?php echo 'give_engraving_message'; ?>" class="postbox">
```
Comment (mehul0810) [inline form-customizations/custom-fields-plugin.php]: @raftaar1191 Please remove PHHP echo here as it is not required in printing ID

Comment (raftaar1191) [reply form-customizations/custom-fields-plugin.php]: Done @mehul0810

**Comment by raftaar1191** [comment]
@mehul0810 PR update as suggested by you

**Comment by raftaar1191** [comment]
Thanks, @mehul0810 for reviewing the PR

@kevinwhoffman can you please assign this issues to someone from the support team so that the doc can be updated?

Here is the docs issue link:
https://github.com/WordImpress/Give-Docs/issues/166

**Pull request merged**